storage/src/test/java/org/apache/kafka/tiered/storage/TieredStorageTestContext.java

```diff
@@ -0,0 +1,339 @@
+            producer.send(record);
+            if (counter++ % batchSize == 0) {
+                producer.flush();
+            }
+        }
+    }
+
+    public List<ConsumerRecord<String, String>> consume(TopicPartition topicPartition,
+                                                        Integer expectedTotalCount,
+                                                        Long fetchOffset) {
+        consumer.assign(Collections.singletonList(topicPartition));
+        consumer.seek(topicPartition, fetchOffset);
+
+        long timeoutMs = 60_000L;
+        String sep = System.lineSeparator();
+        List<ConsumerRecord<String, String>> records = new ArrayList<>();
+        Function1<ConsumerRecords<String, String>, Object> pollAction = polledRecords -> {
+            polledRecords.forEach(records::add);
+            return records.size() >= expectedTotalCount;
+        };
+        Function0<String> messageSupplier = () ->
+                String.format("Could not consume %d records of %s from offset %d in %d ms. %d message(s) consumed:%s%s",
+                        expectedTotalCount, topicPartition, fetchOffset, timeoutMs, records.size(), sep,
+                        Utils.join(records, sep));
+        TestUtils.pollRecordsUntilTrue(consumer, pollAction, messageSupplier, timeoutMs);
+        return records;
+    }
+
+    public Long nextOffset(TopicPartition topicPartition) {
+        List<TopicPartition> partitions = Collections.singletonList(topicPartition);
+        consumer.assign(partitions);
+        consumer.seekToEnd(partitions);
+        return consumer.position(topicPartition);
+    }
+
+    public Long beginOffset(TopicPartition topicPartition) {
+        List<TopicPartition> partitions = Collections.singletonList(topicPartition);
+        consumer.assign(partitions);
+        consumer.seekToBeginning(partitions);
+        return consumer.position(topicPartition);
+    }
+
+    public void bounce(int brokerId) {
+        KafkaBroker broker = brokers.apply(brokerId);
+        closeClients();
+        broker.shutdown();
+        broker.awaitShutdown();
+        broker.startup();
+        initContext();
+    }
+
+    public void stop(int brokerId) {
+        KafkaBroker broker = brokers.apply(brokerId);
+        closeClients();
+        broker.shutdown();
+        broker.awaitShutdown();
+        initContext();
+    }
+
+    public void start(int brokerId) {
+        KafkaBroker broker = brokers.apply(brokerId);
+        closeClients();
+        broker.startup();
+        initContext();
+    }
+
+    public void eraseBrokerStorage(int brokerId) throws IOException {
+        localStorages.get(brokerId).eraseStorage();
+    }
+
+    public TopicSpec topicSpec(String topicName) {
+        synchronized (topicSpecs) {
+            return topicSpecs.get(topicName);
+        }
+    }
+
+    public LocalTieredStorageSnapshot takeTieredStorageSnapshot() {
+        return LocalTieredStorageSnapshot.takeSnapshot(tieredStorages.get(0));
```

Comment:
I assume that we want this snapshot from broker(0) because the RemoteStorage data will remain same no matter which RSM you call from? In that case, I will suggest to change "0" to take a snapshot from "alive" broker. List of "alive" brokers is maintained by KafkaServerTestHarness